src/execution/operator/schema/physical_create_table.cpp

```diff
@@ -7,21 +7,45 @@
 
 namespace duckdb {
 
-void PhysicalCreateTable::GetChunkInternal(ExecutionContext &context, DataChunk &chunk, PhysicalOperatorState *state) {
-	int64_t inserted_count = 0;
+PhysicalCreateTable::PhysicalCreateTable(LogicalOperator &op, SchemaCatalogEntry *schema,
+                                         unique_ptr<BoundCreateTableInfo> info)
+    : PhysicalSink(PhysicalOperatorType::CREATE, op.types), schema(schema), info(move(info)) {
+}
 
+//===--------------------------------------------------------------------===//
+// Sink
+//===--------------------------------------------------------------------===//
+class CreateTableGlobalState : public GlobalOperatorState {
+public:
+	CreateTableGlobalState() {
+	}
+	ChunkCollection materialized_child;
+};
+
+unique_ptr<GlobalOperatorState> PhysicalCreateTable::GetGlobalState(ClientContext &context) {
+	return make_unique<CreateTableGlobalState>();
+}
+
+void PhysicalCreateTable::Sink(ExecutionContext &context, GlobalOperatorState &state, LocalSinkState &lstate_,
+                               DataChunk &input) {
+	lock_guard<mutex> client_guard(append_lock);
+	auto &sink = (CreateTableGlobalState &)state;
+	sink.materialized_child.Append(input);
```

Comment:
I think we can directly append to the base table here, no? That would avoid needing to do a full materialization of the table. We can simply keep a counter of how many elements we appended and return that in the `GetChunkInternal`.

Reply:
The table is created when `GetChunkInternal` is called, it does not exist yet when data is sunk into there. Creating the when `GetGlobalState` is called is problematic, because this operator is also used when `CREATE TABLE` is used, in which case there is no pipeline, and `GetGlobalState` is not called.

At least, I think. Is there a way around this?

Reply:
Perhaps these could be two separate operators, i.e. `PhysicalCreateTable` and `PhysicalCreateTableAs`?

Reply:
I split them up, which made the code much cleaner. Good call! Waiting for tests.